src/Layouts/Main/TechVideos/index.jsx

```diff
@@ -82,10 +136,9 @@ function TechVideos({ initialPageContent, introText }) {
 		// eslint-disable-next-line
 	}, []);
 
-	// set videos from Firebase
-	const dataFromIntro = (val) => {
-		setosValue(val);
-	};
+	useEffect(() => {
+		RetrieveVideos(dataFromFirebase, setVideoValue);
+	});
```

Comment:
Without the dependency array, this code would run not only run on first on mount but also on every subsequent rerender regardless of why the rerender happened. Is this what you intended? otherwise, pass in an empty dependency array

Refer this: https://stackoverflow.com/a/58579462/6922149